### PR TITLE
refactor alerting storage apis

### LINKED ISSUE #938: Refactor Storage APIs & Secrets

### PR DESCRIPTION
**Pull request by alexandreLamarre**
Closes #938

### CODE REVIEW

File: pkg/apis/alerting/v1/alerting.proto [hunk 1]
```diff
@@ -29,13 +29,13 @@ message AlertCondition {
...
-  SilenceInfo silence = 7;
-  // bool smoothing = 8;
-  // https://github.com/rancher/opni-metrics-service/blob/main/metric-forecasting/metric_anomaly_detector.py#L154
-  // bool anomalyDetection = 9;
+
+  // the following are read-only
+  SilenceInfo silence = 7;// Set & Unset via the silence API
+  string id = 8;
+  string goldenSignal = 9;
+  string overrideType = 10;
```
Comment (kralicky) [inline pkg/apis/alerting/v1/alerting.proto:38]: It doesn't seem obvious at first glance what these fields are for, maybe add comments here describing them?

File: pkg/apis/alerting/v1/alerting.proto [hunk 2]
```diff
@@ -431,6 +431,9 @@ message AlertEndpoint {
     EmailEndpoint email = 5;
     PagerDutyEndpoint pagerDuty = 6;
   }
+
+  // the following are read only
+  string id = 7;
```
Comment (kralicky) [inline pkg/apis/alerting/v1/alerting.proto:436]: this seems weird

File: pkg/apis/alerting/v1/alerting.proto [hunk 3]
```diff
@@ -575,4 +578,19 @@ message EphemeralDispatcherRequest {
 
 message EphemeralDispatcherResponse{
   TriggerAlertsRequest triggerAlertsRequest = 1;
+}
+
+message CachedState {
+  bool healthy = 1; 
+  bool firing = 2;
```
Comment (kralicky) [inline pkg/apis/alerting/v1/alerting.proto:585]: Are these mutually exclusive?

Comment (alexandreLamarre) [reply pkg/apis/alerting/v1/alerting.proto:585]: In the backend when the internal evaluator uses these :

`firing = T = > healthy = F`
`healthy = T = > firing = F`
However the converse :
`healthy = F =/=> firing = T`

I'll add a comment on the proto to clarify this

File: pkg/apis/alerting/v1/alerting.proto (hunk 2 above)
Comment (alexandreLamarre) [reply pkg/apis/alerting/v1/alerting.proto:436]: *should be considered read only by any consumer of the API

File: pkg/apis/alerting/v1/alerting.proto (hunk 1 above)
Comment (alexandreLamarre) [reply pkg/apis/alerting/v1/alerting.proto]: Added descriptions:
- golden signal : the kubernetes golden signal this condition falls under
- overrideType : overrides the AlertTypeDetails.String() type for use in the UI

**Comment by alexandreLamarre** [comment]
Closing in favor of https://github.com/rancher/opni/pull/942

**Pull request closed without merging**